tests/test_config.py

```diff
@@ -91,3 +93,22 @@ def test_can_read_toml_rich_types(tmp_path):
 """
     )
     check_configuration_file_reader(config_file)
+
+
+def test_can_read_env_variable(tmp_path):
+    # Check that we can read the "regular" INI .pylintrc file
+    # if it has an environment variable.
+    os.environ["tmp_path_env"] = str(tmp_path)
+    config_file = (
+        Path(os.path.expandvars(os.path.expanduser("${tmp_path_env}"))) / ".pylintrc"
```

Comment:
I think we should not expand the value in the test, to be able to see that the extends is properly done inside the function ? What am I missing ?